pkg/serviceprovider/access_token_mapper.go

```diff
@@ -63,14 +66,24 @@ func (at AccessTokenMapper) ToSecretType(secretType corev1.SecretType) map[strin
 			at.Token)
 	case corev1.SecretTypeSSHAuth:
 		ret[corev1.SSHAuthPrivateKey] = at.Token
+	default:
+		at.fillByMapping(mapping, ret)
 	}
 
 	return ret
 }
 
-// FillByMapping sets the data from the mapper into the provided map according to the settings specified in the provided
+// fillByMapping sets the data from the mapper into the provided map according to the settings specified in the provided
 // mapping.
-func (at AccessTokenMapper) FillByMapping(mapping *api.TokenFieldMapping, existingMap map[string]string) {
+func (at AccessTokenMapper) fillByMapping(mapping *api.TokenFieldMapping, existingMap map[string]string) {
+
+	// if there are no mapping.Token - add "token" key
+	if mapping.Token == "" {
```

Comment:
I would only add the `token` "default mapping" if there is no other mapping. I.e. the user may choose not to put the token into the secret and we should not disallow that IMHO. Here, I'd just add an `IsEmpty()` method on the TokenFieldMapping checking that all the fields are empty and use that in this method.

```go
if mapping.IsEmpty() {
  existingMap[tokenKey] = at.Token
  return
}

// the rest of the method unchanged from before this PR
```

Reply:
@metlos is there any sense in secret which does not contain token? 
I guessed that's the main reason of this issue to not to make it happen.